cdap-ui/app/directives/my-popover/my-popover.js

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+angular.module(PKG.name + '.commons')
+  .directive('myPopover', function($compile, $popover, $timeout) {
+    return {
+      restrict: 'A',
+      scope: {
+        template: '=',
+        contentData:'=',
+        title: '@',
+        placement: '@',
+        popoverContext: '='
+      },
+      link: function(scope, element) {
+        scope.isOpen = false;
+        var popoverElement;
+        var delayOpenTimer;
+        var delayCloseTimer;
+        var mypopover;
+
+        var targetElement = angular.element(element);
+        targetElement.removeAttr('my-popover');
+        targetElement.removeAttr('data-template');
+        targetElement.removeAttr('data-placement');
+        targetElement.removeAttr('data-title');
```

Comment:
General cleanup? Those attributes don't have to be there when we render the ui-bootstrap popup underneath. Might be confusing for someone to see two different attributes with same value. Also ```my-popover``` has to be removed otherwise we will get infinite recursion.